components/raftstore/src/store/worker/read.rs

```diff
@@ -1153,6 +1169,47 @@ where
                             }
                         }
                     }
+                    RequestPolicy::ReadIndex => {
```

Comment:
The following `try_local_stale_read` is mainly used for follower read and has little effect on read index for leader read. How about separating the follower read path in the request policy, then the code would be like:
```
match self.pre_propose_raft_command(&req) {
   Ok(Some((mut delegate, policy))) => {
        match policy {
            RequestPolicy::ReadLocal => ...
            RequestPolicy::StaleRead => ...
            RequestPolicy::ReadIndexReplicaRead => ...  // New for follower read only
         }
   }
   Ok(None) => self.redirect(RaftCommand::new(req, cb)), // Or try the optimization too for leader read
}
```